src/PowerShellEditorServices/Analysis/AnalysisService.cs

```diff
@@ -467,24 +455,73 @@ private PSObject[] InvokePowerShell(string command, IDictionary<string, object>
             return await task;
         }
 
-        private bool VerifyPSScriptAnalyzerAvailable()
+        /// <summary>
+        /// Create a new runspace pool around a PSScriptAnalyzer module for asynchronous script analysis tasks.
+        /// This uses the PowerShell module API to load the first PSScriptAnalyzer module on the module path with
+        /// with the required minimum version and then creates a runspace pool with that module loaded in the initial
+        /// session state.
+        /// </summary>
+        /// <returns>A runspace pool with PSScriptAnalyzer loaded for running script analysis tasks.</returns>
+#if PowerShellv5
```

Comment:
Not sure if that'll work like you expect unfortunately.  We don't actually build separate assemblies for each version (except as a test), the compiler directives just to allow it to compile.

The way we get this to work in the debugger is by having an interface (`IVersionSpecificOperations`) and several implementations for each PowerShell version required.  Then at runtime the implementation for that version is picked.   That way no missing member exceptions get thrown.

Here's the version specific interface stuff

[IVersionSpecificOperations](https://github.com/PowerShell/PowerShellEditorServices/blob/4bb71bd93320ba9b2105e3b5d2007f09f0f4a959/src/PowerShellEditorServices/Session/IVersionSpecificOperations.cs)
[PowerShell3Operations](https://github.com/PowerShell/PowerShellEditorServices/blob/4bb71bd93320ba9b2105e3b5d2007f09f0f4a959/src/PowerShellEditorServices/Session/PowerShell3Operations.cs)
[PowerShell4Operations](https://github.com/PowerShell/PowerShellEditorServices/blob/4bb71bd93320ba9b2105e3b5d2007f09f0f4a959/src/PowerShellEditorServices/Session/PowerShell4Operations.cs)
[PowerShell5Operations](https://github.com/PowerShell/PowerShellEditorServices/blob/4bb71bd93320ba9b2105e3b5d2007f09f0f4a959/src/PowerShellEditorServices/Session/PowerShell5Operations.cs)

And here's where that's used in [PowerShellContext](https://github.com/PowerShell/PowerShellEditorServices/blob/4bb71bd93320ba9b2105e3b5d2007f09f0f4a959/src/PowerShellEditorServices/Session/PowerShellContext.cs#L227-#L245).

At least, I'm pretty sure that's how it works.  It's possible that was done for readability though.  If y'all have a PSv3 machine there I'd give it a go real quick to confirm.  If it crashes immediately, that's probably why.

Reply:
Bugger! I was just basing it on the defined constants in the csproj. I'll see if I can test it on a v3 machine. And then ideally we should save the better solution for later somewhere (not that I'm necessarily suggesting leaving it in as a comment)